src/create-theme-provider.js

```diff
@@ -0,0 +1,92 @@
+const React = require('react');
+const PropTypes = require('prop-types');
+const isFunction = require('is-function');
```

Comment:
Because it fixes an ancient issue with RegExp which is older than IE9, just tested typeof new RegExp('') returns an object.

A use case where theme provider is called from an iframe is also irrelevant, if you go this route you would need to fix a lot more.